addons/account/controllers/request_document.py

```diff
@@ -0,0 +1,35 @@
+from markupsafe import Markup
+
+from odoo import http
+from odoo.http import request
+
+
+class AccountRequestDocument(http.Controller):
+
+    @http.route(
+        '/account/request_document/<token>',
+        type='http', auth='public', website=True)
+    def request_document(self, token=None):
+        document = request.env['account.requested.document'].sudo().search([('token', '=', token)])
```

Comment:
With your current access rights and declaration of the `token` field,  you don't need the `sudo()` here and same in the other method (except if portal users should be able to use the route ?) :+1: But once you'll have added the `groups=` to the token field, you might need it depending on who you want to allow to search the documents through this route. Please reconsider the `sudo()` :smile:

Reply:
Actually, the whole point of requesting the document through a link sharing is to enable public access to the users who don't have access to the accounting app to upload their missing documents 
for example: the accountant can send an email to his customer to say Hello customer I don't have the invoices for your Microsoft 360 subscription or I don't have the details of your Visa Card.

So, in that case, using sudo() is mandatory in both requesting and upload the document.